src/org/labkey/test/components/ui/search/FilterExpressionPanel.java

```diff
@@ -92,11 +92,19 @@ public void setFilter(int index, Object operator, Object value1, Object value2)
             }
             else if (value1 instanceof Date dateVal1)
             {
-                final String dateStr1 = DATE_FORMAT.format(dateVal1);
+                String dateStr1;
+                if (value1 instanceof DateString)
+                    dateStr1 = ((DateString)value1).getDateStr();   // use the test-supplied date string
```

Comment:
New Java syntax for `instanceof` to avoid casting
```suggestion
                if (value1 instanceof DateString ds1)
                    dateStr1 = ds1.getDateStr();   // use the test-supplied date string
```